controllers/machineset_controller.go

```diff
@@ -384,6 +384,12 @@ func (r *MachineSetReconciler) getNewMachine(machineSet *clusterv1.MachineSet) *
 	machine.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, machineSetKind)}
 	machine.Namespace = machineSet.Namespace
 	machine.Spec.ClusterName = machineSet.Spec.ClusterName
+
+	machine.Labels[clusterv1.MachineSetLabelName] = machineSet.Name
+	if md := metav1.GetControllerOf(machineSet); md != nil {
+		machine.Labels[clusterv1.MachineDeploymentLabelName] = md.Name
```

Comment:
Can't we apply this when creating the MS in machinedeployment_controller?

Reply:
I think we can do that, but I'm do not quite understand why we need to do that. Obviously I'm missing the context/background here :(

Reply:
Might be good to do both? A MachineSet might not be managed by a MachineDeployment

Reply:
@vincepri @detiber i've updated the pr, PTAL if it looks right :) thanks!